src/main/steps/c100-rebuild/check-your-answers/mainUtil.ts

```diff
@@ -1068,42 +1068,49 @@ const RespondentDetails_AddressAndPersonal = (sessionRespondentData, respondent,
-          changeUrl: applyParms(Urls['C100_RESPONDENT_DETAILS_CONTACT_DETAILS'], { respondentId: id }),
-        },
-      );
-    }else {
-      newRespondentStorage.push(
-      {
-        key:(language === 'en')? 'Email': getYesNoTranslation(language,'Email', 'personalDetails'),
-            value: contactDetails?.['emailAddress'],
-            changeUrl: applyParms(Urls['C100_RESPONDENT_DETAILS_CONTACT_DETAILS'], { respondentId: id }),
-      },
-    );}
-   
-    if(contactDetails.hasOwnProperty('donKnowTelephoneNumber') && contactDetails['donKnowTelephoneNumber'] === 'Yes'){
-      newRespondentStorage.push(
-        {
-          key: (language === 'en')? 'I dont know their telephone number':getYesNoTranslation(language,'I dont know their telephone number', 'personalDetails'),
-          value: getYesNoTranslation(language,contactDetails?.['donKnowTelephoneNumber'],'doTranslation'),
-          changeUrl: applyParms(Urls['C100_RESPONDENT_DETAILS_CONTACT_DETAILS'], { respondentId: id }),
-        },
-      );
-    }else{
-      newRespondentStorage.push(
-        {
-          key: (language === 'en')? 'Telephone number' :getYesNoTranslation(language,'Telephone number', 'personalDetails'),
-          value: contactDetails?.['telephoneNumber'],
-          changeUrl: applyParms(Urls['C100_RESPONDENT_DETAILS_CONTACT_DETAILS'], { respondentId: id }),
-        }
-      );
-    }
+  newRespondentStorage.push(respondentEmailDetails(contactDetails, id, language));
+  newRespondentStorage.push(respondentTelephoneDetails(contactDetails, id, language));
+
+  return newRespondentStorage;
+};
+
+const respondentEmailDetails = (contactDetails, id, language) => {
+  if (contactDetails.hasOwnProperty('donKnowEmailAddress') && contactDetails['donKnowEmailAddress'] === 'Yes') {
+    return {
+      key:
+        language === 'en'
+          ? 'I dont know their email address'
+          : getYesNoTranslation(language, 'I dont know their email address', 'personalDetails'),
+      value: getYesNoTranslation(language, contactDetails?.['donKnowEmailAddress'], 'doTranslation'),
+      changeUrl: applyParms(Urls['C100_RESPONDENT_DETAILS_CONTACT_DETAILS'], { respondentId: id }),
+    };
+  } else {
+    return {
+      key: language === 'en' ? 'Email' : getYesNoTranslation(language, 'Email', 'personalDetails'),
+      value: contactDetails?.['emailAddress'],
+      changeUrl: applyParms(Urls['C100_RESPONDENT_DETAILS_CONTACT_DETAILS'], { respondentId: id }),
+    };
+  }
+};
 
-    return newRespondentStorage;
+const respondentTelephoneDetails = (contactDetails, id, language) => {
+  if (contactDetails.hasOwnProperty('donKnowTelephoneNumber') && contactDetails['donKnowTelephoneNumber'] === 'Yes') {
+    return {
+      key:
+        language === 'en'
+          ? 'I dont know their telephone number'
+          : getYesNoTranslation(language, 'I dont know their telephone number', 'personalDetails'),
```

Comment:
Do not have english phrases as keys to translations. come up with meaningful short phrases joined with an _. for eg: I dont know their telephone number can have the key as dont_know_telephone